pyartcd/pyartcd/oc.py

```diff
@@ -35,3 +39,47 @@ async def registry_login(runtime: Runtime):
     except ChildProcessError:
         runtime.logger.error('Failed to login into OC registry')
         raise
+
+
+def get_release_image_pullspec(release_pullspec: str, image: str):
+    # oc adm release info --image-for=<image> <pullspec>
+    with oc.tracking() as tracker:
+        try:
+            r = Result("adm")
+            r.add_action(oc.oc_action(oc.cur_context(), 'adm', cmd_args=['release', 'info', f'--image-for={image}', release_pullspec]))
+            if r.status() == 0:
+                logger.info(f"Output: {r.out().strip()}")
+            else:
+                logger.warn(r.err())
+        except Exception as e:
+            logger.error(tracker.get_result())
+            raise e
+    return r.status(), r.out().strip()
+
+
+def extract_release_binary(image_pullspec: str, path_args: list[str]):
+    # oc image extract --confirm --only-files --path=/usr/bin/..:<workdir> <pullspec>
+    with oc.tracking() as tracker:
+        try:
+            r = Result("image")
+            r.add_action(oc.oc_action(oc.cur_context(), 'image', cmd_args=['extract', '--confirm', '--only-files'] + path_args + [image_pullspec]))
+            r.fail_if("extract release binary failed")
+        except Exception as e:
+            logger.error(tracker.get_result())
+            raise e
+
+
+def extract_release_client_tools(release_pullspec: str, path_arg: str, arch: str):
+    # oc adm release extract --tools --command-os='*' -n ocp --to=<workdir> --filter-by-os=<arch> --from <pullspec>
+    with oc.tracking() as tracker:
+        try:
+            r = Result("tools")
```

Comment:
Curious, what does "tools" mean in this context?

Reply:
it's a self-defined operation name https://github.com/openshift/openshift-client-python/blob/main/packages/openshift/result.py#L8